src/Engines/Nunjucks.js

```diff
@@ -4,6 +4,78 @@ const TemplatePath = require("../TemplatePath");
 const EleventyErrorUtil = require("../EleventyErrorUtil");
 const EleventyBaseError = require("../EleventyBaseError");
 
+/*
+ * HACKITYHACKHACKHACKHACK
+ */
+let inc = ((i) => {
+  return () => {
+    return i++;
+  };
+})(0);
+let pathMap = new Map();
+
+let clearCache = (reason = "") => {
+  if (pathMap.size > 0) {
+    pathMap.clear();
```

Comment:
This clears the entire map?
Did you meant to clear only the `reason` part?

Reply:
This clears the entire map; WIP versions used reason to log why things were getting clobbered as an assist to track down correctness issues. Will remove the `reason` flag entirely.